### PR TITLE
Add hosting of openapi.json file.

### PR DESCRIPTION
**Pull request by chambridge**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,8 +101,9 @@ create-test-db-file: run-migrations
 
 gen-apidoc:
 	rm -fr $(PYDIR)/staticfiles/
+	rm -fr $(APIDOC)
 	apidoc -i $(PYDIR) -o $(APIDOC)
-	cp docs/source/specs/openapi.yml $(APIDOC)/
```
Comment (dccurtis) [inline Makefile]: Need a `mkdir $(APIDOC)`?

I hit this 
```
(koku_master-2goqrU2u) ➜  koku_master git:(646-serve-openapi) make gen-apidoc
rm -fr koku/staticfiles/
rm -fr apidoc
apidoc -i koku -o apidoc
2019-02-19 14:08:08,155 - ERROR - Failed to parse sources
cp docs/source/specs/openapi.json apidoc/
cp: directory apidoc does not exist
make: *** [gen-apidoc] Error 1
```

Even after this it's still failing for me.  Am I doing something wrong?
```
(koku_master-2goqrU2u) ➜  koku_master git:(646-serve-openapi) ✗ make gen-apidoc
rm -fr koku/staticfiles/
rm -fr apidoc
mkdir apidoc
apidoc -i koku -o apidoc
2019-02-19 14:13:21,719 - ERROR - Failed to parse sources
cp docs/source/specs/openapi.json apidoc/
(koku_master-2goqrU2u) ➜  koku_master git:(646-serve-openapi) ✗
```

Comment (dccurtis) [reply Makefile]: Also adding ApiDoc to Pipfile would be great

Comment (chambridge) [reply Makefile]: Strange I get the following:
```
$ make gen-apidoc
rm -fr koku/staticfiles/
rm -fr apidoc
apidoc -i koku -o apidoc
info: Done.
cp docs/source/specs/openapi.json apidoc/
```

[apidoc](http://apidocjs.com/) is NodeJS so can't add it to the Pipfile.

Comment (dccurtis) [reply Makefile]: 🤦‍♂️  ok that explains it :)

Comment (chambridge) [reply Makefile]: Not your fault its only kinda documented through the code install. I'll open a bug to doc that part.

Comment (chambridge) [reply Makefile]: https://github.com/project-koku/koku/issues/651

File: Makefile [hunk 2]
```diff
@@ -101,8 +101,9 @@ create-test-db-file: run-migrations
...
+	cp docs/source/specs/openapi.json $(APIDOC)/
 
 collect-static:
```
Comment (dccurtis) [inline Makefile]: First time for me going through this openapi stuff so bear with me.  I had to run `make collect-static` as well to get the `/api/v1/openapi.json` endpoint to return anything.  Should this be part of the `gen-apidoc` Makefile definition?

File: koku/koku/dev_middleware.py [hunk 3]
```diff
@@ -39,11 +39,11 @@ def process_request(self, request):  # pylint: disable=no-self-use
         if hasattr(request, 'META'):
             identity_header = {'identity':
                                {
-                                   'account_number': '10001',
+                                   'account_number': '1460290',
                                    'type': 'User',
                                    'user': {
-                                       'username': 'user_dev',
-                                       'email': 'user_dev@foo.com'
+                                       'username': 'jalberts@redhat.com',
```
Comment (dccurtis) [inline koku/koku/dev_middleware.py]: Mistake??

Comment (chambridge) [reply koku/koku/dev_middleware.py]: yup 🤦‍♂️

File: Makefile (hunk 2 above)
Comment (chambridge) [reply Makefile]: I suppose we could make `gen-apidoc` a target that gets run when you execute `make collect-static`. 🤷‍♂️

Comment (dccurtis) [reply Makefile]: Maybe.  I'd say whatever would make sure the right thing happens for our OpenShift deployment.  I just wanted to call out that you get a 404 if `make collect-static` isn't ran after generating the api doc.  Wasn't clear what the problem was until I started it up with pdb

**Comment by chambridge** [comment]
@dccurtis http://editor.swagger.io/ allows you to save it as JSON. Also converts it back to yaml when you copy the JSON into the editor window.

File: Makefile (hunk 2 above)
Comment (chambridge) [reply Makefile]: yeah the openshift does it we are already serving the apidocs and yaml up til now
https://github.com/project-koku/koku/blob/master/openshift/s2i/bin/assemble#L98

**Comment by dccurtis** [comment]
Cool good to know!

**Pull request merged**